apps/frontend/src/components/template/PdfTemplateEditorViewer.tsx

```diff
@@ -0,0 +1,60 @@
+import { Viewer } from '@react-pdf-viewer/core';
+import { Worker } from '@react-pdf-viewer/core';
+import React, { useCallback, useContext } from 'react';
+
+import '@react-pdf-viewer/core/lib/styles/index.css';
+
+import { UserContext } from 'context/UserContextProvider';
+import { PdfTemplate } from 'generated/sdk';
+
+function PdfTemplateEditorViewer({
+  pdfTemplate,
+}: {
+  pdfTemplate: PdfTemplate;
+}) {
+  const { token } = useContext(UserContext);
+  const [generatedPdfPreviewBlob, setGeneratedPdfPreviewBlob] =
+    React.useState<Blob>();
+  const fetchGeneratedPdfPreviewData = useCallback(async () => {
+    const pdf = await fetch(
+      `/preview/pdf/proposal?pdfTemplateId=${pdfTemplate.pdfTemplateId}`,
+      {
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${token}`,
+        },
+      }
+    );
+    const pdfBlob = await pdf.blob();
+
+    return pdfBlob;
+  }, [pdfTemplate, token]);
+
+  React.useEffect(() => {
+    fetchGeneratedPdfPreviewData().then((pdfBlob) => {
+      setGeneratedPdfPreviewBlob(pdfBlob);
+    });
+  }, [fetchGeneratedPdfPreviewData]);
+
+  return (
+    <Worker workerUrl="https://unpkg.com/pdfjs-dist@3.4.120/build/pdf.worker.min.js">
```

Comment:
I see that in the `package.json` you added this library as dependency `pdfjs-dist` but I don't understand why we need to load it from an external source.